lib/controllers/european.controller.ts

```diff
@@ -41,7 +41,44 @@ export class EuropeanController extends SessionController {
   }
 
   public async refreshAccessToken(): Promise<string> {
-    return this.login();
+    const shouldRefreshToken = Math.floor(Date.now() / 1000 - this.session.tokenExpiresAt) >= -10;
+
+    if (!this.session.refreshToken) {
+      return Promise.reject('Need refresh token to refresh access token. Use login()');
+    }
+
+    if (!shouldRefreshToken) {
+      return 'Token not expired, no need to refresh';
```

Comment:
This should return a `Promise.resolve(...)`

Reply:
As you can read [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function#Description) it's semantically equivallent. So do you want this as a code style?
However then I already introduced that here: 
https://github.com/Hacksore/bluelinky/blob/8fcdff4c95debf385e0accef2865a3b23c426be8/lib/vehicles/european.vehicle.ts#L293-L296
One could also argue, that when you are already dealing with async functions why do you need Promise methods. The above part would become:
```js
        return 'Start charge successful';
    }

    throw 'Something went wrong!';
```
But obviously it should be consistent in the libary.

Reply:
Excellent point, since this is only code style I'd not want you to take on the debt of fix it elsewhere.

Since there nothing different from a functional standpoint I'm ok with this.